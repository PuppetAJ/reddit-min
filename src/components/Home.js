import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { fetchAllPosts, selectAllPosts } from "../features/Posts/postsSlice";

function Home() {
  const dispatch = useDispatch();
  const posts = useSelector(selectAllPosts);

  useEffect(() => {
    dispatch(fetchAllPosts());
    console.log(posts);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  // const fetchAll = async () => {
  //   const data = await fetch("https://www.reddit.com/r/all/top.json");
  //   console.log(await data.json());
  // };

  return <h1>Home</h1>;
}

export default Home;
