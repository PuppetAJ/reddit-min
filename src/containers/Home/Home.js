import React, { useEffect } from "react";
import { useAppDispatch, useAppSelector } from "../../app/hooks";
import { fetchPosts, selectAllPosts } from "../../features/Posts/postsSlice";
import PostsList from "../../features/Posts/PostsList";

function Home() {
  const dispatch = useAppDispatch();
  const posts = useAppSelector(selectAllPosts);

  useEffect(() => {
    dispatch(fetchPosts({ sub: "all", mode: "hot" }));
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <div>
      <h1>Home</h1>
      <PostsList postsData={posts} />
    </div>
  );
}

export default Home;
