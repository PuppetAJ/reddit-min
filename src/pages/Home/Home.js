import React, { useEffect } from "react";
import { useAppDispatch, useAppSelector } from "../../app/hooks";
import { fetchPosts, selectAllPosts } from "../../features/Posts/postsSlice";
import {
  fetchSubreddits,
  selectAllSubreddits,
} from "../../features/Subreddits/subredditsSlice";
import PostsList from "../../features/Posts/PostsList";
import SubredditsList from "../../features/Subreddits/SubredditsList";

function Home() {
  const dispatch = useAppDispatch();
  const posts = useAppSelector(selectAllPosts);
  const subreddits = useAppSelector(selectAllSubreddits);

  useEffect(() => {
    dispatch(fetchPosts({ sub: "all", mode: "hot" }));
    dispatch(fetchSubreddits());
    // eslint-disable-next-line
  }, []);

  // const asyncFetch = async () => {
  //   const data = await fetch("https://www.reddit.com/best/communities/1.json/");
  //   console.log(await data.json());
  // };

  setTimeout(() => {
    console.log(posts);
    console.log(subreddits);
  }, 1000);

  return (
    <div className="home-container">
      <h1>Home</h1>
      <div>
        <PostsList postsData={posts} />
        <SubredditsList subredditsData={subreddits} />
      </div>
    </div>
  );
}

export default Home;
