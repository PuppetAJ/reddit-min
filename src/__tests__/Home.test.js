import { render, screen, cleanup, waitFor } from "@testing-library/react";
import { Provider } from "react-redux";
import Home from "../pages/Home/Home";
import { configureStore } from "@reduxjs/toolkit";
import postsReducer from "../features/Posts/postsSlice";
import renderer from "react-test-renderer";

const mockStore = configureStore({
  reducer: {
    posts: postsReducer,
  },
});

afterEach(() => {
  cleanup();
});

describe("Home container", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("renders without crashing", () => {
    const storeSpy = jest.spyOn(mockStore, "dispatch");

    render(
      <Provider store={mockStore}>
        <Home />
      </Provider>
    );

    const component = renderer.create(
      <Provider store={mockStore}>
        <Home />
      </Provider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(screen.getByText("Home")).toBeInTheDocument();
    expect(storeSpy).toHaveBeenCalledTimes(1);
    expect(storeSpy).toHaveBeenCalledWith(expect.any(Function));
  });

  it("dispatches fetchPosts action", async () => {
    render(
      <Provider store={mockStore}>
        <Home />
      </Provider>
    );
    await waitFor(async () => {
      expect(await screen.findByText("0")).toBeInTheDocument();
      expect(
        await screen.findByText("Title: PUBG: BATTLEGROUNDS")
      ).toBeInTheDocument();
      expect(await screen.findByText("Home")).toBeInTheDocument();
    });
  });
});
