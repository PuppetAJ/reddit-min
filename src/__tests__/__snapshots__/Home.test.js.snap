// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home container renders without crashing 1`] = `
<div>
  <h1>
    Home
  </h1>
  <div>
    <h2>
      Posts
    </h2>
    <ul />
  </div>
</div>
`;
